docs/_docs/reference/contextual/multiversal-equality.md

```diff
@@ -27,19 +27,19 @@ But it will probably give unexpected results and fail at runtime.
 Multiversal equality is an opt-in way to make universal equality safer.
 It uses a binary type class [`scala.CanEqual`](https://github.com/lampepfl/dotty/blob/main/library/src/scala/CanEqual.scala)
 to indicate that values of two given types can be compared with each other.
-The example above would not typecheck if `S` or `T` was a class
-that derives `CanEqual`, e.g.
+The example above would not typecheck if there was a given `CanEqual[S,S]` or `CanEqual[T,T]` in scope but no a `CanEqual[T,S]`. For example:
```

Comment:
```suggestion
The example above would not typecheck if there was a given `CanEqual[S,S]` or `CanEqual[T,T]` in scope but would if there was a given `CanEqual[T,S]`. For example:
```

Reply:
```suggestion
If you add a given `CanEqual[S,S]` or `CanEqual[T,T]` to the above (and in scope), it stops compiling since `S` and `T` cannot be compared, the code compiles again if there is a given `CanEqual[T,S]` added (and in scope). An example of such would be:
```

Reply:
For me, the original is most clear. (Except it should use subjunctive "if `S` or `T` were a class that derives".) The possible improvement is changing "would not typecheck" to "We can fix the example by changing S or T so that it derives CanEqual. That prevents the equals test from typechecking." And I would add to the example, `x == y // no longer typechecks`. I like that it presents `derives` first, then `given`, then how to supply givens that make equality work again. However, that last example should use `S` and `T` instead of `A` and `B`, for continuity. (I like the progression in the text, but I need a few more words of hand-holding so I don't lose the thread.)

I think your suggested changes reflect that you know how it works already.

Reply:
For me the `derives` is a shorthand for not having to use `given`, whereas you see it in the other way, `given` being an desugaring detail of `derives`

I had not considered this point of view, but I think it makes more sense to talk about `given`s first and foremost since we need them anyways for `CanEqual[T,S]`

> However, that last example should use S and T instead of A and B, for continuity.

Which example ?

> reflect that you know how it works already
I didn't before reading this page, so my changes are written from that perspective, having to ask around for what really goes on

Reply:
derives is not really just short hand for given - because it makes it globally available due to being in the companion object

Reply:
Hmm, that's true, in the end, I think @som-snytt is correct, I'll revert and add some more details, notably a link to type-class derivation